### PR TITLE
Build/fix no prefix

### LINKED ISSUE #1086: Asset paths not generated correctly for schemeless external urls
Asset paths get clobbered when `EXTERNAL_URL` is missing the protocol scheme.

### PR DESCRIPTION
**Pull request by conectado**
Fixes #1086 #1073 

For #1073 the problem was docker doing dnat changing the destination IP and without a HOST header that would cause a cert problem (No cert for the dnatted IP) and a no site route (the route was generated for the original destination IP not the dnatted one). The fix was to put the container in `network_mode: "host"` so no dnat ocurred.

For #1086 the the problem was that if you didn't add `https://` to the `EXTERNAL_URL` it'd cause a bug where the `ip` or `host` was taken as a path instead of `host` when using `URI.parse` which in turn would cause the Route helpers to generate the static routes incorrectly. The solution is trying to detect if it's a correctly generated url and if not add the `https://` .

### CODE REVIEW

File: apps/fz_common/lib/fz_net.ex [hunk 1]
```diff
@@ -65,4 +65,12 @@ defmodule FzCommon.FzNet do
   def valid_hostname?(hostname) when is_binary(hostname) do
     String.match?(hostname, @host_regex)
   end
+
+  def to_complete_url(str) when is_binary(str) do
+    case URI.parse(str) do
+      %{host: nil, scheme: nil} -> "https://" <> str
+      %{scheme: nil} -> "https" <> str
```
Comment (jamilbk) [inline apps/fz_common/lib/fz_net.ex]: Out of curiosity, when does this happen?

Comment (conectado) [reply apps/fz_common/lib/fz_net.ex]: In an edge case a user types `//hostname`

Comment (jamilbk) [reply apps/fz_common/lib/fz_net.ex]: I see -- maybe we shouldn't boot up if `EXTERNAL_URL` is invalid.

Comment (conectado) [reply apps/fz_common/lib/fz_net.ex]: Oh just did some testing, caddy doesn't support that so will remove it

**Review by jamilbk — APPROVED** [review]

**Pull request merged**